twingate/provider.go

```diff
@@ -19,13 +21,19 @@ const (
 	DefaultURL          = "twingate.com"
 
 	// EnvAPIToken env var for Token.
-	EnvAPIToken     = "TWINGATE_API_TOKEN" //#nosec
-	EnvNetwork      = "TWINGATE_NETWORK"
-	EnvURL          = "TWINGATE_URL"
-	EnvHTTPTimeout  = "TWINGATE_HTTP_TIMEOUT"
-	EnvHTTPMaxRetry = "TWINGATE_HTTP_MAX_RETRY"
+	EnvAPIToken      = "TWINGATE_API_TOKEN" //#nosec
+	EnvNetwork       = "TWINGATE_NETWORK"
+	EnvURL           = "TWINGATE_URL"
+	EnvHTTPTimeout   = "TWINGATE_HTTP_TIMEOUT"
+	EnvHTTPMaxRetry  = "TWINGATE_HTTP_MAX_RETRY"
+	EnvCorrelationID = "TWINGATE_CORRELATION_ID"
 )
 
+func init() { //nolint:gochecknoinits
+	correlationID, _ := uuid.GenerateUUID()
+	_ = os.Setenv(EnvCorrelationID, correlationID)
```

Comment:
Why do we have different parts of our code communicating a value via OS env var?

This is not a good way and allows interfering with it from the outside

Reply:
Initially I tried to init this value directly in http client constructor, but when I run some acc test, I’ve noticed that this id was changing from request to request.I can test some other approach as well to not use env var, but even if someone defines the same env var we will overwrite it.

Reply:
I will be back online in a few hours, and can try another approach

Reply:
ok, changed to global var